libethcore/Miner.cpp

```diff
@@ -1,19 +1,101 @@
+/*
+ This file is part of ethereum.
+
+ ethminer is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ ethereum is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ethminer.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 #include "Miner.h"
-#include "EthashAux.h"
 
-using namespace dev;
-using namespace eth;
+namespace dev
+{
+namespace eth
+{
+unsigned Miner::s_dagLoadMode = 0;
+
+unsigned Miner::s_dagLoadIndex = 0;
+
+unsigned Miner::s_dagCreateDevice = 0;
+
+uint8_t* Miner::s_dagInHostMemory = nullptr;
+
+bool Miner::s_exit = false;
+
+bool Miner::s_noeval = false;
+
+double Miner::s_alpha = 1.0;
+
+std::ostream& operator<<(std::ostream& os, HwMonitor _hw)
+{
+    os << _hw.tempC << "C " << _hw.fanP << "%";
+    if (_hw.powerW)
+        os << ' ' << fixed << setprecision(0) << _hw.powerW << "W";
+    return os;
+}
+
+std::ostream& operator<<(std::ostream& os, FormattedMemSize s)
+{
+    static const char* suffixes[] = {"bytes", "KB", "MB", "GB"};
+    double d = s.m_size;
```

Comment:
@jean-m-cyr: trivial warning on Windows here

```
C:\Users\xmr\Desktop\ethminer\libethcore\Miner.cpp(49): warning C4244: 'initializing': conversion from 'uint64_t' to
'double', possible loss of data [C:\Users\xmr\Desktop\ethminer\build\libethcore\ethcore.vcxproj]
```

Reply:
Easy fix: double d = double(s.m_size);

Submit a PR if you want.